### DISCUSSION TITLE
stepper timeout (Laser module Ender3-V2 NEO)

### DISCUSSION THREAD

**Discussion by andreas-fe** [mriscoc/Ender3V2S1#938] [Ideas]
First of all,m I´m very satisfied with this firmware and the provided options.
I also have the aditionsl Creality laser module for engraving and cutting with my 3D printer.

For the Ender3-V2 NEO this is an aditional control unit, which is connected directly to the X and Y axis of the printer, and the z-cable is connected to the laser.

The Problem is, that the Z-Axis needs to be fixed, at a specific high.
Herefore, I used the function to move the Z-Axis a little bit up, and then the stepper for the z-axis isfixed, to keep the heavy laser in position.
The problem here is, that there is a stepper timeout, to disable the stepper automatically.
I found the section "Idle Stepper Shutdown" in the Configuration_adv.h with the possibility to deactivatze the timeout for the z-axis.
But I´m not sure, if this have any side effect.
Is it OK to change the value, or is it also possible to add this parameter (change Stepper Timeout) in the config menue in a further version?

**Comment by andreas-fe** [comment]
Or another idea,
In the comment of the configuration file is the note, that it is also possible, to change the timeout with M18 and M84 Commands. Set to 0 for No Timeout

Is it also possible to send this command to the printer, with an gcode file?
I tried to send a shrinked gcode file with the command "M84 Z E S0", but in my case this was not working.
I also tried some tests with a minimal set og gcodes, but at the end, all steppers were disabled.
But maybe only my syntax was wrong.

**Reply by mriscoc** [reply]
On Ender printer stock boards, all axes share the same activation pin, so it is not possible keep Z enable while X and Y are disabled.

**Reply by andreas-fe** [reply]
Thank you for your feedback.
I tested a working solution. I´ve created a config foile with the G-code to set Z-Stepper Timeout to unlimited:
After the appliing, I only need to move zhe Z-axis.

example
```gcode
; Professional Firmware Configuration File
; Config to disable Z-Axis Timeout
;=====================================================
C10                    ; Mark as a configuration file
M117 Applying configuration
;-----------------------------------------------------
M84 Z E S0      	   ; Set Z-Stepper Timeout to unlimited
M500                   ; Save all settings
;-----------------------------------------------------
G4 S1                  ; Wait a second
M300 P200              ; Beep
M117 Configuration Applied
;-----------------------------------------------------
```
